engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/MessageTransformer.java

```diff
@@ -34,7 +38,7 @@ public void transform(final Message element, final TransformContext context) {
     final ExecutableMessage executableElement = new ExecutableMessage(id);
     final ExtensionElements extensionElements = element.getExtensionElements();
 
-    if (extensionElements != null) {
+    if (extensionElements != null && !isReferredByStartEvent(element)) {
```

Comment:
:x: The `isReferredByStartEvent` method solves the issue by preventing the `if` block from being executed by validating that the current `message` isn't referred by a `start event`.

This is a bit redundant, and the `isReferredByStartEvent` is a duplication of [this message validation](https://github.com/camunda/zeebe/blob/20c87ed36587fc6e73b55e97de19868a12ced4ca/bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/MessageValidator.java#L77-L90) code.

:thought_balloon: We already validate against the two Message use-cases in the `MessageValidator` class, i.e. 
* If the message is referred by a Catch event, Receive task, or Event SubProcess start event, we check for a `name` attribute and for a `subscription` element.
* If the message is referred by a process Start event, we only check for a `name` attribute.

So, when we get to the `MessageTransformer`, we have already validated the use cases, so it would be better to avoid another validation check.

:wrench: I think we can do this by expanding the [Query](https://github.com/camunda/zeebe/blob/dbac8504f4cf680865c44919252f0c3118db44e2/bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/Query.java#L39) API, by adding a new method there. The method can be `Optional<T> findSingleResult()`.

Using this method, we can then get the `correlationKeyExpression` only when a `subscription` is available.

With this refactoring, we will have two benefits. We will improve the Query API, we will avoid code duplication. And we will generalize the `MessageTransformer` code so that the specific use-cases are not as visible here.